src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java

```diff
@@ -303,6 +306,46 @@ public static JsonArrayBuilder json(List<DataverseContact> dataverseContacts) {
         }
         return jsonArrayOfContacts;
     }
+    
+    public static JsonArrayBuilder getOwnersFromDvObject(DvObject dvObject) {
+        
+        List <DvObject> ownerList = new ArrayList();
+        dvObject = dvObject.getOwner(); // We're going to ignore the object itself
+        while (dvObject != null) {
+            ownerList.add(dvObject);
+            dvObject = dvObject.getOwner();
+        } 
+        
+        JsonArrayBuilder jsonArrayOfOwners = Json.createArrayBuilder();
+        
+        for (DvObject dvo : ownerList){
+                JsonObjectBuilder ownerObject = jsonObjectBuilder();
+                if (dvo.isInstanceofDataverse()){
+                    ownerObject.add("type", "DATAVERSE");
+                }
+                if (dvo.isInstanceofDataset()){
+                    ownerObject.add("type", "DATASET");
+                }
+                 if (dvo.isInstanceofDataFile()){
+                    ownerObject.add("type", "DATAFILE");
+                }
+                if (dvo.isInstanceofDataverse()){
+                    Dataverse in = (Dataverse) dvo;
+                    ownerObject.add("identifier", in.getAlias());
+                }
+                if (dvo.isInstanceofDataset() || dvo.isInstanceofDataFile() ){
+                    if (dvo.getIdentifier() != null){
+                        Dataset ds = (Dataset) dvo;
```

Comment:
I can get here if dvo is a DataFile and it has a global ID right? so the cast might fail. I don't think it is needed though.

Reply:
Fixed the logic so that it doesn't need to cast or decide whether there's a dataset or datafile - basically for the purposes of "isPartOf" we're ignoring the object itself so we're never dealing with files.